mkosi/__init__.py

```diff
@@ -5008,6 +4987,10 @@ def run_verb(args: Args, tools: Optional[Config], images: Sequence[Config], *, r
     # The images array has been modified so we need to reevaluate last again.
     last = images[-1]
 
+    if args.verb == Verb.bump:
+        finalize_image_version(args, last)
```

Comment:
Should this still check if `mkosi.version` is executable? Even worse, if it doesn't check but overwrite the file content it is still marked executable and the next invocation of mkosi will try to execute it (which fails)

Reply:
We check if it's executable during config parsing

Reply:
But the `finalize_image_version` unconditionally writes the version string into it even if it is executable or am I missing something? I.e. `mkosi bump` with an executable `mkosi.version` just breaks now?

Reply:
Ah I think I get it now, please forget it